openml_OS/models/api/v1/Api_data.php

```diff
@@ -294,23 +327,31 @@ private function data_edit() {
       $this->returnError( 1064, $this->version );
       return;
     }
-    // check if user owns dataset
-    if($dataset->uploader != $this->user_id and !$this->user_has_admin_rights) {
-      $this->returnError( 1065, $this->version );
-      return;
+
+    // If critical fields need to be edited
+    if (isset($update_fields['default_target_attribute']) || isset($update_fields['row_id_attribute']) || isset($update_fields['ignore_attribute'])) {
```

Comment:
rather than checking these one by one, I would actually prefer to create an "edit XSD", so that we have this consistent in our set of XSD documents (that are easier to review than the code itself)